src/Rules/Italian/RulesetInflector.php

```
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Inflector\Rules\Italian;
+
+use function implode;
+use function preg_split;
+use function strpos;
+
+use const PREG_SPLIT_DELIM_CAPTURE;
+use const PREG_SPLIT_NO_EMPTY;
+
+class RulesetInflector extends \Doctrine\Inflector\RulesetInflector
+{
+    public function inflect(string $word): string
+    {
+        // If it's a single word without spaces or hyphens, use the original inflector
+        if (strpos($word, ' ') === false && strpos($word, '-') === false) {
+            return parent::inflect($word);
+        }
+
+        // Split the phrase into words and process each one
+        $words = preg_split('/([ -])/', $word, -1, PREG_SPLIT_DELIM_CAPTURE | PREG_SPLIT_NO_EMPTY);
+        if ($words === false) {
+            return parent::inflect($word);
```

If this fails, it feels like you should rather throw an exception.

https://3v4l.org/evCNt#vnull

---

I’d say that as a fallback it’s right for it to handle the string as if it were a single word, so I would leave it like that without raising exceptions, because in Italian there are often compound or multiple words, so if there were any unmanaged cases exceptions would be raised, whereas this way it would still be handled.